src/Nimbus/Infrastructure/AzureQueueManager.cs

```diff
@@ -83,12 +85,40 @@ public Task<MessageSender> CreateMessageSender(string queuePath)
 
         public Task<MessageReceiver> CreateMessageReceiver(string queuePath)
         {
+            if (!_hasAzureManageClaim)
+            {
```

Comment:
There's a lot of repetition of this code, it really should be in the EnsureQueueExists / EnsureTopicExists methods